theano/tensor/basic.py

```diff
@@ -744,7 +744,12 @@ def get_scalar_constant_value(orig_v, elemwise=True):
 
 def tensor(*args, **kwargs):
     name = kwargs.pop('name', None)
-    return TensorType(*args, **kwargs).make_variable(name=name)
+    # This add an indirection to the normal call stack. So raise the
+    # limit to keep the good user line.
+    limit = config.traceback.limit
+    if limit != -1:
+        limit += 1
+    return TensorType(*args, **kwargs)(name=name, limit=limit)
```

Comment:
Another way I think of is to raise the default limit by one. Then this would make the stack printed to the user bigger.

Another way that is now possible as the cost of a high limit is much lower now, is to use a bigger limit by default and return only one stack level after trimming the internal of Theano part as we currently do.

Personally, I do not see much harm by this work around. Maybe we can keep it for now and it we need more of this then go for something more dynamic?